File: src/database/entity/deck.ts
import { Entity, PrimaryGeneratedColumn, Column, BaseEntity, ManyToOne, OneToMany } from "typeorm";
import { Account } from "./account";
import { Card } from "./card";
import { GameClass } from "./gameClass";

@Entity({ name: 'deck' })
export class Deck extends BaseEntity {

  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @ManyToOne(() => GameClass, (gameClass: GameClass) => gameClass.decks, { onDelete: 'CASCADE' })
  class: GameClass;

  @ManyToOne(() => Account, (account: Account) => account.decks, { onDelete: 'CASCADE' })
  account: Account;

  @OneToMany(() => Card, (card: Card) => card.deck, {onDelete: "CASCADE"})
  cards: Card[];

}

File: DotnetBackend/RandomCards/Entities/Deck.cs
using System.Collections.Generic;

namespace RandomCards.Entities
{
    public class Deck : BaseEntity
    {
        public int ClassId { get; set; }
        public int AccountId { get; set; }
        public string Name { get; set; }

        public virtual List<Card> Cards { get; set; }
        public virtual Account Account { get; set; }
        public virtual Class Class { get; set; }
    }
}
